diff -Naur a/av.rkt b/av.rkt
--- a/av.rkt	2015-01-11 19:37:21.323476726 -0800
+++ b/av.rkt	2015-01-11 19:40:06.664490379 -0800
@@ -12,9 +12,20 @@
                      _int32_t
                      _uint16_t
                      _uint32_t
-                     _uint64_t))
+                     _uint64_t
+		     exe-path
+		     lib-path))
 
-(define-ffi-definer define-av (ffi-lib "libtoxav"))
+; jenkins junk
+(define exe-path (path->string (find-system-path 'run-file)))
+
+; shenanigans to get the binary to be called from anywhere and still work
+(define lib-path (string->path (string-append
+                                (substring exe-path 0
+                                           (- (string-length exe-path) 6))
+                                "../lib/libtoxav")))
+
+(define-ffi-definer define-av (ffi-lib lib-path))
 
 ; The _string type supports conversion between Racket strings
 ; and char* strings using a parameter-determined conversion.
diff -Naur a/blight.rkt b/blight.rkt
--- a/blight.rkt	2015-01-11 19:37:25.451477067 -0800
+++ b/blight.rkt	2015-01-11 19:41:02.868495020 -0800
@@ -11,9 +11,20 @@
 (provide (except-out (all-defined-out)
                      define-blight
                      _uint8_t
-                     _uint32_t))
+                     _uint32_t
+		     exe-path
+		     lib-path))
 
-(define-ffi-definer define-blight (ffi-lib "libblight"))
+; jenkins junk
+(define exe-path (path->string (find-system-path 'run-file)))
+
+; shenanigans to get the binary to be called from anywhere and still work
+(define lib-path (string->path (string-append
+                                (substring exe-path 0
+                                           (- (string-length exe-path) 6))
+                                "../lib/libblight")))
+
+(define-ffi-definer define-blight (ffi-lib lib-path))
 
 (define _uint8_t _uint8)
 (define _uint32_t _uint32)
diff -Naur a/dns.rkt b/dns.rkt
--- a/dns.rkt	2015-01-11 19:37:28.708477336 -0800
+++ b/dns.rkt	2015-01-11 19:42:25.045501805 -0800
@@ -8,9 +8,20 @@
                      define-dns
                      _uint8_t
                      _uint16_t
-                     _uint32_t))
+                     _uint32_t
+		     exe-path
+		     lib-path))
 
-(define-ffi-definer define-dns (ffi-lib "libtoxdns"))
+; jenkins junk
+(define exe-path (path->string (find-system-path 'run-file)))
+
+; shenanigans to get the binary to be called from anywhere and still work
+(define lib-path (string->path (string-append
+                                (substring exe-path 0
+                                           (- (string-length exe-path) 6))
+                                "../lib/libtoxdns")))
+
+(define-ffi-definer define-dns (ffi-lib lib-path))
 
 (define _uint8_t _uint8)
 (define _uint16_t _uint16)
diff -Naur a/encrypt.rkt b/encrypt.rkt
--- a/encrypt.rkt	2015-01-11 19:37:33.932477768 -0800
+++ b/encrypt.rkt	2015-01-11 19:43:15.581505978 -0800
@@ -7,9 +7,20 @@
 (provide (except-out (all-defined-out)
                      define-encrypt
                      _Tox-pointer
-                     _uint32_t))
+                     _uint32_t
+		     exe-path
+		     lib-path))
 
-(define-ffi-definer define-encrypt (ffi-lib "libtoxencryptsave"))
+; jenkins junk
+(define exe-path (path->string (find-system-path 'run-file)))
+
+; shenanigans to get the binary to be called from anywhere and still work
+(define lib-path (string->path (string-append
+                                (substring exe-path 0
+                                           (- (string-length exe-path) 6))
+                                "../lib/libtoxencryptsave")))
+
+(define-ffi-definer define-encrypt (ffi-lib lib-path))
 
 #|###################
  # type definitions #
diff -Naur a/functions.rkt b/functions.rkt
--- a/functions.rkt	2015-01-11 19:37:43.101478525 -0800
+++ b/functions.rkt	2015-01-11 19:43:55.524509276 -0800
@@ -10,9 +10,20 @@
                      _uint8_t
                      _uint16_t
                      _uint32_t
-                     _uint64_t))
+                     _uint64_t
+		     exe-path
+		     lib-path))
 
-(define-ffi-definer define-tox (ffi-lib "libtoxcore"))
+; jenkins junk
+(define exe-path (path->string (find-system-path 'run-file)))
+
+; shenanigans to get the binary to be called from anywhere and still work
+(define lib-path (string->path (string-append
+                                (substring exe-path 0
+                                           (- (string-length exe-path) 6))
+                                "../lib/libtoxcore")))
+
+(define-ffi-definer define-tox (ffi-lib lib-path))
 
 #|###################
  # type definitions #
