diff -Naur a/av.rkt b/av.rkt
--- a/av.rkt	2015-05-25 23:03:59.056008085 -0700
+++ b/av.rkt	2015-05-25 23:05:27.785017144 -0700
@@ -9,12 +9,16 @@
 (provide (except-out (all-defined-out)
                      define-av
                      _Tox-pointer
+		     exe-path
                      libtoxav-path))
 
+(define exe-path (path->string (find-system-path 'run-file)))
+
 (define libtoxav-path
   (if (eq? (system-type) 'windows)
       "libtox"
-      "libtoxav"))
+      (string-append (substring exe-path 0 (- (string-length exe-path) 6))
+		     "../lib/libtoxav")))
 
 (define-ffi-definer define-av (ffi-lib libtoxav-path))
 
diff -Naur a/dns.rkt b/dns.rkt
--- a/dns.rkt	2015-05-25 23:03:59.056008085 -0700
+++ b/dns.rkt	2015-05-25 23:05:27.785017144 -0700
@@ -7,12 +7,16 @@
 (provide (except-out (all-defined-out)
                      define-dns
                      TOX_PUBLIC_KEY_SIZE
+		     exe-path
                      libtoxdns-path))
 
+(define exe-path (path->string (find-system-path 'run-file)))
+
 (define libtoxdns-path
   (if (eq? (system-type) 'windows)
       "libtox"
-      "libtoxdns"))
+      (string-append (substring exe-path 0 (- (string-length exe-path) 6))
+		     "../lib/libtoxdns")))
 
 (define-ffi-definer define-dns (ffi-lib libtoxdns-path))
 
diff -Naur a/encrypt.rkt b/encrypt.rkt
--- a/encrypt.rkt	2015-05-25 23:03:59.056008085 -0700
+++ b/encrypt.rkt	2015-05-25 23:05:27.785017144 -0700
@@ -8,12 +8,16 @@
 (provide (except-out (all-defined-out)
                      define-encrypt
                      _Tox-pointer
+		     exe-path
                      libtoxencryptsave-path))
 
+(define exe-path (path->string (find-system-path 'run-file)))
+
 (define libtoxencryptsave-path
   (if (eq? (system-type) 'windows)
       "libtox"
-      "libtoxencryptsave"))
+      (string-append (substring exe-path 0 (- (string-length exe-path) 6))
+		     "../lib/libtoxencryptsave")))
 
 (define-ffi-definer define-encrypt (ffi-lib libtoxencryptsave-path))
 
diff -Naur a/functions.rkt b/functions.rkt
--- a/functions.rkt	2015-05-25 23:03:59.056008085 -0700
+++ b/functions.rkt	2015-05-25 23:05:49.368019348 -0700
@@ -7,13 +7,17 @@
          "enums.rkt")
 (provide (except-out (all-defined-out)
                      define-tox
+		     exe-path
                      libtoxcore-path
                      _Bool))
 
+(define exe-path (path->string (find-system-path 'run-file)))
+
 (define libtoxcore-path
   (if (eq? (system-type) 'windows)
       "libtox"
-      "libtoxcore"))
+      (string-append (substring exe-path 0 (- (string-length exe-path) 6))
+		     "../lib/libtoxcore")))
 
 (define-ffi-definer define-tox (ffi-lib libtoxcore-path))
 
